### PR TITLE
Run scripts/fix-headers.py

### PR DESCRIPTION
**Pull request by EnricoMi**
Adds `encoding="utf-8"` to `open` in `scripts/fix-headers.py` as suggested by @trim21 in #2801: https://github.com/PyGithub/PyGithub/pull/2857/files#diff-5ea215bdc91826fad536021a63ef5013a1f9ed7410ced16df2ef00432f45722b

Then runs `scripts/fix-headers.py`.

### CODE REVIEW

File: github/Branch.py [hunk 1]
```diff
@@ -9,9 +9,22 @@
...
 # Copyright 2018 Steve Kowalik <steven@wedontsleep.org>                        #
 # Copyright 2018 Wan Liuyang <tsfdye@gmail.com>                                #
 # Copyright 2018 sfdye <tsfdye@gmail.com>                                      #
+# Copyright 2019 Steve Kowalik <steven@wedontsleep.org>                        #
+# Copyright 2019 Wan Liuyang <tsfdye@gmail.com>                                #
+# Copyright 2020 Steve Kowalik <steven@wedontsleep.org>                        #
+# Copyright 2021 Mark Walker <mark.walker@realbuzz.com>                        #
+# Copyright 2021 Steve Kowalik <steven@wedontsleep.org>                        #
+# Copyright 2023 Jirka Borovec <6035284+Borda@users.noreply.github.com>        #
+# Copyright 2023 Juan Manuel "Kang" Pérez <kangcoding@gmail.com>               #
```
Comment (EnricoMi) [inline github/Branch.py:22]: @Borda `codespell` complains about `Manuel` here, any chance you can tell `codespell` to stop complaining?

Comment (trim21) [reply github/Branch.py:22]: see my pr

Comment (Borda) [reply github/Branch.py:23]: Yes, add it to exceptions

**Comment by JLLeitschuh** [comment]
Rebased pull request on top of #2858

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
Thanks!

**Pull request merged**